uninstall.php

```diff
@@ -5,64 +5,116 @@
  * @package OneAccess
  */
 
+declare( strict_types=1 );
+
+namespace OneAccess;
+
 // If uninstall not called from WordPress, exit.
 if ( ! defined( 'WP_UNINSTALL_PLUGIN' ) ) {
 	exit;
 }
 
-if ( ! function_exists( 'oneaccess_plugin_sync_deactivate' ) ) {
+/**
+ * Multisite loop for uninstalling from all sites.
+ */
+function multisite_uninstall(): void {
+	if ( ! is_multisite() ) {
+		uninstall();
+		return;
+	}
+
+	delete_network_plugin_data();
+
+	$site_ids = get_sites(
+		[
+			'fields' => 'ids',
+			'number' => 0,
+		]
+	) ?: [];
+
+	foreach ( $site_ids as $site_id ) {
+		// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.switch_to_blog_switch_to_blog
+		if ( ! switch_to_blog( (int) $site_id ) ) {
+			continue;
+		}
+
+		uninstall();
+		restore_current_blog();
+	}
+}
+
+/**
+ * The (site-specific) uninstall function.
+ */
+function uninstall(): void {
+	delete_plugin_data();
+}
+
+/**
+ * Delete multisite network plugin data.
+ */
+function delete_network_plugin_data(): void {
+	$options = [
+		'oneaccess_multisite_governing_site',
+	];
 
-	/**
-	 * Function to deactivate the plugin and clean up options.
-	 */
-	function oneaccess_plugin_sync_deactivate(): void {
+	foreach ( $options as $option ) {
+		delete_site_option( $option );
+	}
+}
 
-		// list of actions to be cleared on uninstall.
-		$actions_to_clear = array(
+/**
+ * Deletes meta, options, transients, etc.
+ */
+function delete_plugin_data(): void {
+
+	// list of actions to be cleared on uninstall.
+		$actions_to_clear = [
 			'oneaccess_governing_site_configured',
 			'oneaccess_add_deduplicated_users',
-		);
+		];
 
 		// Clear scheduled actions.
 		if ( function_exists( 'as_unschedule_all_actions' ) ) {
 			foreach ( $actions_to_clear as $action ) {
 				// check if action is scheduled then clear it.
-				if ( as_next_scheduled_action( $action ) ) {
-					as_unschedule_all_actions( $action );
+				if ( ! as_next_scheduled_action( $action ) ) {
+					continue;
 				}
+
+				as_unschedule_all_actions( $action );
 			}
 		}
 
```

Comment:
The prescheduled check isnt necessary for an uninstall, it's done internally.

```suggestion
		if ( function_exists( 'as_unschedule_all_actions' ) ) {
			foreach ( $actions_to_clear as $action ) {
				as_unschedule_all_actions( $action );
			}
		}

```